flink-ml-lib/src/test/java/org/apache/flink/ml/util/MockKVStore.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/** Class that manages global key-value pairs used in unit tests. */
+@SuppressWarnings({"unchecked"})
+public class MockKVStore {
+    private static final Map<String, Object> map = new HashMap<>();
+
+    private static long counter = 0;
+
+    /**
+     * Returns a prefix string to make sure key-value pairs created in different test cases would
+     * not have the same key.
+     */
+    public static synchronized String createNonDuplicatePrefix() {
```

Comment:
I agree that it would be better to remove this method. But in this case a random int64 value might bring risk, as test cases running in parallel might generate identical keys. I'll try to avoid this problem by using a more complex key pattern, like composing class name and a monotonically increasing int64.

Reply:
In practice it is extremely unlikely that two randomly generated int64 values collide with each other. This is called `statistically unique`.

According to the Javadoc of Flink's JobID, the JobID is statistically unique. And Flink already relies on this kind of uniqueness to differentiate multiple Flink jobs in the same Flink session in production. It seems that there is no need for strict uniqueness in our tests?

Reply:
Got it. As we are using InMemoryReporter and thus won't be able to implement this in MockKVStore, I'll apply this principle to the design of simplified test blocking queue mechanisms.